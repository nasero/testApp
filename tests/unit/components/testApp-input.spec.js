import { shallowMount } from '@vue/test-utils';
import TestAppInput from '@/components/testApp-input';

const directives = {
    validate: jest.fn()
};

describe('TestAppInput - Snapshot', () => {
    it('Snapshot', () => {
        const wrapper = shallowMount(TestAppInput, {
            directives
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('TestAppInput - methods', () => {
    it('btnClicked', async () => {
        const wrapper = shallowMount(TestAppInput, {
            directives
        });

        wrapper.vm.emitValue({
            target: {
                value: 'foo'
            }
        });

        await wrapper.vm.$nextTick();

        expect(wrapper.emitted('testApp-input-value')).toBeTruthy();
    });
});
