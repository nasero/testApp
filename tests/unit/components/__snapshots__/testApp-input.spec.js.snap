// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestAppInput - Snapshot Snapshot 1`] = `
<div
  class="testApp-input"
>
  <label
    class="hidden"
  >
    
  </label>
  <input
    class="testApp-input__input"
    type="text"
  />
  <p
    class="testApp-input__error"
  >
    
  </p>
</div>
`;
